warden/x/async/keeper/keeper.go

```diff
@@ -289,3 +289,47 @@ func (k Keeper) callEVMWithData(
 
 	return res, nil
 }
+
+func (k Keeper) getCompletedFuturesWithoutValidatorVote(ctx context.Context, valAddress []byte, limit int) ([]prophet.FutureResult, error) {
+	it, err := k.futures.results.IterateRaw(ctx, nil, nil, collections.OrderDescending)
+	if err != nil {
+		return nil, err
+	}
+	defer it.Close()
+
+	futures := make([]prophet.FutureResult, 0, limit)
+	for ; it.Valid(); it.Next() {
+		id, err := it.Key()
+		if err != nil {
+			return nil, err
+		}
+
+		result, err := it.Value()
+		if err != nil {
+			return nil, err
+		}
+
+		if found, _ := k.votes.Has(ctx, collections.Join(id, valAddress)); found {
+			continue
+		}
```

Comment:
we shouldn't ignore the error, sorry for being misleading in my previous comment

```suggestion
		found, err := k.votes.Has(ctx, collections.Join(id, valAddress))
		if found {
			continue
		}
		if err != nil {
			return nil, err
		}
```

Reply:
That would be strange if `found` was `true` and `err` wansn't `nil`, though

Reply:
any sort of I/O error could happen while reading the db, plus an error is also returned in case the key (valAddress) is not of the correct format. It shouldn't happen, but we also shouldn't ignore it